lettre/src/smtp/extension.rs

```diff
@@ -163,6 +168,11 @@ impl ServerInfo {
             };
         }
 
+        // If the server did not send any `AUTH` features, assume Plain is supported
+        if !supports_any_auth {
```

Comment:
I think we should not support `PLAIN` arbitrarily everywhere, but instead just ignore the response from the server when `force_set_auth` is true.